### PR TITLE
 ♻️🔒 Use unique ids for UserTask events

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Refactor the tests to use InstanceIDs for finishing UserTasks, instead of ModelIDs.

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/126

PR: #150

## How can others test the changes?

Run the tests.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: _integration_tests/test/user_task_tests.js [hunk 1]
```diff
@@ -97,15 +99,24 @@ describe('User Tasks - ', () => {
     await processInstanceHandler.startProcessInstanceAndReturnCorrelationId(processModelId, correlationId, initialToken);
     await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId);
 
+    let waitingUserTasks = await processInstanceHandler.getWaitingUserTasksForCorrelationId(identity, correlationId);
+    const userTask1 = waitingUserTasks.userTasks[0];
```
Comment (robpal94) [inline _integration_tests/test/user_task_tests.js]: We should ensure here, that the `waitingUserTask` variable contains an array and has at least one element.

File: _integration_tests/test/user_task_tests.js [hunk 2]
```diff
@@ -97,15 +99,24 @@ describe('User Tasks - ', () => {
...
     };
 
-    await processInstanceHandler.finishUserTaskInCorrelation(identity, correlationId, processModelId, 'User_Task_1', userTaskInput);
+    await processInstanceHandler
+      .finishUserTaskInCorrelation(identity, correlationId, userTask1.processInstanceId, userTask1.flowNodeInstanceId, userTaskInput);
     await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId);
-    await processInstanceHandler.finishUserTaskInCorrelation(identity, correlationId, processModelId, 'User_Task_2', userTaskInput);
+
+    waitingUserTasks = await processInstanceHandler.getWaitingUserTasksForCorrelationId(identity, correlationId);
+    const userTask2 = waitingUserTasks.userTasks[0];
```
Comment (robpal94) [inline _integration_tests/test/user_task_tests.js]: See above

File: _integration_tests/test/user_task_tests.js [hunk 3]
```diff
@@ -194,6 +203,9 @@ describe('User Tasks - ', () => {
     await processInstanceHandler.startProcessInstanceAndReturnCorrelationId(processModelId, correlationId, initialToken);
     await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId);
 
+    const waitingUserTasks = await processInstanceHandler.getWaitingUserTasksForCorrelationId(identity, correlationId);
+    const userTask = waitingUserTasks.userTasks[0];
```
Comment (robpal94) [inline _integration_tests/test/user_task_tests.js]: See above

**Comment by ElRaptorus** [comment]
@Vyperus @robpal94  done

**Pull request merged**